docs/WEBXDC.md

```diff
@@ -0,0 +1,232 @@
+- Real-time data via dedicated IPC channels
+- File operations through secure IPC bridges
+
+## Bootstrap Sequence
+
+1. **App Launch**: User clicks on Webxdc message
+2. **Window Creation**: Main process creates isolated BrowserWindow
+3. **Protocol Registration**: Custom `webxdc://` handler is set up for this session
+4. **Wrapper Loading**: Browser loads `webxdc_wrapper.html` from protocol handler
+5. **Preload Injection**: `webxdc-preload.js` is injected and sets up the API bridge
+6. **API Setup**: `webxdc.js` script initializes the API with app-specific data
+7. **App Loading**: The iframe loads the actual Webxdc app (`index.html`)
+8. **Ready State**: App is fully loaded and can interact with the Webxdc API
+
+## Security Model
+
+### Multi-Layer Isolation
+
+1. **Process Isolation**: Each Webxdc app runs in its own renderer process
+2. **Session Isolation**: Apps have separate storage, cookies and cache (per account)
+3. **Context Isolation**: Preload script runs in isolated context
+4. **Iframe Sandboxing**: Additional iframe isolation within the renderer
+5. **CSP Enforcement**: Strict Content Security Policy on all responses
+
+### Permission Model
+
+Webxdc apps have extremely limited permissions:
+
+- **Allowed**: Fullscreen, pointer lock (for games)
+- **Blocked**: Camera, microphone, geolocation, notifications, etc.
+- **Network**: Controlled per-app (some apps can access internet via maps:// protocol)
+
+### Content Security Policy
+
+```javascript
+const CSP =
+  "default-src 'self';\
+  style-src 'self' 'unsafe-inline' blob: ;\
+  font-src 'self' data: blob: ;\
+  script-src 'self' 'unsafe-inline' 'unsafe-eval' blob: ;\
+  connect-src 'self' data: blob: ;\
+  img-src 'self' data: blob: ;\
+  media-src 'self' data: blob: ;\
+  webrtc 'block'"
+```
+
+## Real-time Communication
+
+WebRTC is disabled by setting WebRTCIPHandlingPolicy to 'disable_non_proxied_udp' for each BrowserWindow. This should force any WebRTC request to use the dummy proxy provided in [webxdc.ts](../packages/target-electron/src/deltachat/webxdc.ts#L123) (see comments [here](../packages/target-electron/src/deltachat/webxdc.ts#L325))
+
+Webxdc apps can communicate in real-time through:
+
+- **Status Updates**: Persistent state changes shared between all app instances
+- **Realtime Channel**: Ephemeral binary data for real-time features
+- **Chat Integration**: Ability to send messages to the hosting chat
+
+## File System Access
+
+Webxdc apps access files through secure channels:
+
+- **Blob Storage**: Files are stored in Delta Chat's blob storage
+- **Import/Export**: Secure file picker and drag-out functionality
+- **Temporary Files**: Secure temporary file creation for exports
+
+## Internet Access
+
+Webxdc apps can't access the internet with one exception:
+
+- **Controlled Access**: Only integrated and explicitly allowed apps get internet access (granted by core)
+- **CSP Relaxation**: Internet-enabled apps get modified CSP for external resources
```

Comment:
It says "one exception" and then lists 2 things. It seems that the intent was to have the "maps.xdc" paragraph go right after "one exception".